### PR TITLE
⚡ add statemine to teleport bridge

### LINKED ISSUE #6024: Teleport KSM to Statemine
Tasks:
- [ ] move /teleport page under /_prefix/teleport

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #6024
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [x] My fix has changed UI

### <samp>🤖 Generated by Copilot at c2d0105</samp>

This pull request enables cross-chain asset teleportation between Kusama and Statemine using the identity store. It refactors the `Teleport.vue` component and adds new types and functions to the `teleport.ts` and `chains.ts` files.

### <samp>🤖 Generated by Copilot at c2d0105</samp>

> _To teleport assets with ease_
> _We refactored the `Teleport.vue` piece_
> _We used the identity store_
> _And simplified the core_
> _And added `chainPrefixesMap` for each `Prefix`_

## Leftovers

- [ ] make balance in teleport page reactive (i.e should change after sending token is done)
* #6227

### CODE REVIEW

**Comment by daiagi** [comment]
#### All these work:
Kusama -> statemine
Stateine -> Kusama
Statemie -> Basilisk
Basilisk -> Kusama
Basilisk -> Statemine

#### What doesn't work

Kusama / Statmine -> Basilisk

* https://github.com/paraspell/xcm-sdk/issues/40

On top of that
- [ ]  make the balance in teleport page reactive

File: components/teleport/Teleport.vue [hunk 1]
```diff
@@ -98,31 +103,27 @@ import { txCb } from '@/utils/transactionExecutor'
...
-const getKusamaApi = async () =>
-  await ApiPromise.create({
-    provider: new WsProvider(getChainEndpointByPrefix('kusama') as string),
-  })
-const getBasiliskApi = async () =>
-  await ApiPromise.create({
-    provider: new WsProvider(getChainEndpointByPrefix('basilisk') as string),
+const getApi = (from: Chain) => {
+  return ApiPromise.create({
+    provider: new WsProvider(getChainEndpointByPrefix(chainToPrefixMap[from])),
```
Comment (vikiival) [inline components/teleport/Teleport.vue:113]: Please check @kodadot1/subapi

File: components/teleport/Teleport.vue [hunk 2]
```diff
@@ -140,13 +141,51 @@ const fromTabs = [
     label: Chain.BASILISK,
     value: Chain.BASILISK,
   },
+  {
+    label: Chain.STATEMINE,
+    value: Chain.STATEMINE,
+  },
 ]
+const ToTabs = [
```
Comment (vikiival) [inline components/teleport/Teleport.vue:149]:
```suggestion
const toTabs = [
```

**Review by vikiival — DISMISSED** [review]
please move teleport to _prefix/teleport

File: components/teleport/Teleport.vue (hunk 1 above)
Comment (daiagi) [reply components/teleport/Teleport.vue]: done

**Comment by daiagi** [comment]
> please move teleport to _prefix/teleport

Done

**Comment by daiagi** [comment]
along the way, PR to fix issue in Paraspell when teleporting from Kusama to Basilisk:

* https://github.com/paraspell/xcm-sdk/pull/41

**Comment by daiagi** [comment]
Ready to review but... 
Attempt to send to Basilisk causes bug in paraspell
I've already submitted pr to close that (see comment above)

Until it get merged we can disable to Basilisk tab

**Comment by Jarsen136** [comment]
I just tried to teleport ksm from Basilisk to Statemine and it looks failed.

tx: https://kusama.subscan.io/xcm_message/kusama-5f15dc6856d44aeba8a9aff8c3f85e6a5572627b

image

**Comment by prury** [comment]
Here we go: 

KSM -> BSX - never asked for signing(I'm aware that it has been already identified) 
KSM -> STATEMINE - Success - https://kusama.subscan.io/tx/0x3a10ae4fd7ca2c1a29fdadb9180b7c4e7ba49ea0e5039751ef95bbd4c87d6540

BSX -> KSM - Success - https://basilisk.subscan.io/tx/0x6fa8971ca05b934b3233d751aeaab6130f0e95c491d124b86d12a0d6d29da1a8
BSX -> STATEMINE - Success - https://basilisk.subscan.io/tx/0xd1e402a93a342a7748baa30d0c65339c299a1fd080ea4fe50f947cfe6c3fac7a

STATEMINE -> KSM - Success - https://statemine.subscan.io/tx/0x1e86518f38f3acad024143ee3952c64ca8d8e0b56e7a7d13fad811784a3efb17
STATEMINE -> BSX - never asked for signing

Small visual bug("usd" being overlayed)

https://github.com/kodadot/nft-gallery/assets/36627808/67208d0b-d2b8-470e-9521-af265d420530

**Comment by daiagi** [comment]
Bumped new version of paraspell
now all works excpet 
Statemine => Basilisk
* https://github.com/paraspell/xcm-sdk/issues/42#issue-1740265754

**Comment by daiagi** [comment]
> I just tried to teleport ksm from Basilisk to Statemine and it looks failed.


all i can see is that tranfers to Statemine take a little bit longer to complete
other then that seems working for me
https://basilisk.subscan.io/block/0xcd9fa8deb8672d8adb27e98b420479eac548c1529614dbe6597fd10ebf6f84eb

**Comment by Jarsen136** [comment]
>  BSX -> STATEMINE - Success - https://basilisk.subscan.io/tx/0xd1e402a93a342a7748baa30d0c65339c299a1fd080ea4fe50f947cfe6c3fac7a
> 

It may not be related to this PR, but I still cannot teleport ksm from bsx to statemine. Even for this success example, it failed at the step 2.

https://basilisk.subscan.io/xcm_message/kusama-d9bc59d1d8dcdcab650f6533128142f366553b5e
image


Here is my another example: https://statemine.subscan.io/xcm_message/kusama-14e877d1ead722d3582439929280ced2023d21a3

**Review by roiLeo — APPROVED** [review]
✅ code lgtm

**Comment by daiagi** [comment]
so as it stands teleporting between Basilisk and statemine on both direction doesn't work
* https://github.com/paraspell/xcm-sdk/issues/42#issuecomment-1581360718

I'll disable these direction and then it is mergable from my prespective

**Comment by prury** [comment]
> so as it stands teleporting between Basilisk and statemine on both direction doesn't work
> 
>     * [unable to teleport statemine => Basilisk paraspell/xcm-sdk#42 (comment)](https://github.com/paraspell/xcm-sdk/issues/42#issuecomment-1581360718)
> 
> 
> I'll disable these direction and then it is mergable from my prespective

Cool, i will do a simple UI check after you disable it

**Comment by daiagi** [comment]
@prury 

you can check
image

**Comment by prury** [comment]
> @prury
> 
> you can check image

- Strange, disabled options are not greyed out here for me:

image

- I still can send from funds from both basilisk and statemine if i left them selected on some other chain and then change to statemine for example: 

https://github.com/kodadot/nft-gallery/assets/36627808/b7e3f14d-63da-4a7f-b409-6f2905d2c978

**Comment by daiagi** [comment]
Yep. I obviously didn't test thoroughly enough.
Thanks @prury

Ill get it fixed

**Review by vikiival — COMMENTED** [review]
sooo let's go?

**Comment by yangwao** [comment]
> still jet lagged

https://www.garmin.com/en-US/garmin-technology/health-science/jet-lag-adviser/

https://www.youtube.com/watch?v=pY4Hkih6E1E

haven't tried it but have it! I would just travel to Asia to test this feature tbh.

**Comment by daiagi** [comment]
@prury please check again. should be good to go

File: libs/ui/src/components/NeoButton/NeoButton.scss [hunk 3]
```diff
@@ -114,12 +114,15 @@
     }
   }
 
-  &--unlockable {
-    height: 40px;
+  &--secondary{
+    height: 22px;
+    line-height: 100%;
+    font-size: 12px;
```
Comment (roiLeo) [inline libs/ui/src/components/NeoButton/NeoButton.scss]: px?

File: libs/ui/src/components/NeoButton/NeoButton.scss [hunk 4]
```diff
@@ -114,12 +114,15 @@
     }
   }
 
-  &--unlockable {
-    height: 40px;
+  &--secondary{
+    height: 22px;
```
Comment (roiLeo) [inline libs/ui/src/components/NeoButton/NeoButton.scss]: fixed height 😦 
can't you use padding instead?

File: libs/ui/src/components/NeoButton/NeoButton.scss (hunk 3 above)
Comment (daiagi) [reply libs/ui/src/components/NeoButton/NeoButton.scss]: not my changes.
they come form a merge
check out this commit:

https://github.com/kodadot/nft-gallery/commit/0943d2ea573f1427895f5e98b85e8fff290b5702#diff-6cac81a09ae16a212a724a6359cce50c6031ffb32b5746b2fc6c34c2164e93e0

from this PR:

* https://github.com/kodadot/nft-gallery/pull/6199

Comment (roiLeo) [reply libs/ui/src/components/NeoButton/NeoButton.scss]: oh ☹️ sorry about that

**Comment by prury** [comment]
> @prury please check again. should be good to go

works fine! ty!

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.32 USD/DOT ~ 11.574 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x533a95b16fa4e71b6fd30b027d158e565e9777d4bc9a86246ab8ea7234768ade](https://polkadot.subscan.io/extrinsic/0x533a95b16fa4e71b6fd30b027d158e565e9777d4bc9a86246ab8ea7234768ade)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**